test/unittest/statistics/rtps/RTPSStatisticsTests.cpp

```diff
@@ -41,182 +43,406 @@ namespace rtps {
 
 struct MockListener : IListener
 {
-    MOCK_METHOD1(on_statistics_data, void(
-                const Data&));
+    void on_statistics_data(
+            const Data& d) override
+    {
+        switch (d._d())
+        {
+            case RTPS_SENT:
+                on_rtps_sent(d.entity2locator_traffic());
+                break;
+            case HEARTBEAT_COUNT:
+                on_heartbeat_count(d.entity_count());
+                break;
+            case ACKNACK_COUNT:
+                on_acknack_count(d.entity_count());
+                break;
+            case DATA_COUNT:
+                on_data_count(d.entity_count());
+                break;
+        }
```

Comment:
I would suggest adding a default case to prevent gcc warnings:

```
warning: enumeration value ‘NETWORK_LATENCY’ not handled in switch [-Wswitch]
```
and so on with the other `EventKind`s not handled.